### PR TITLE
Complete Password Generator UI

### PR DESCRIPTION
**Pull request by Asterisk-z**
# Description

I designed the password generator ui and it works fine without bug but when i updated my folked repository and pull it into my mechine i get this error aside that my implementation is working great
Screenshot from 2022-05-30 03-06-39

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by atapas** [comment]
> # Description
> I designed the password generator ui and it works fine without bug but when i updated my folked repository and pull it into my mechine i get this error aside that my implementation is working great Screenshot from 2022-05-30 03-06-39
> 
> Fixes # (issue)
> 
> ## Type of change
> * [x]  New feature (non-breaking change which adds functionality)
> 
> # How Has This Been Tested?
> Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
> 
> # Checklist:
> * [x]  I have performed a self-review of my own code
> * [x]  I have commented my code, particularly in hard-to-understand areas
> * [ ]  I have made corresponding changes to the documentation
> * [ ]  My changes generate no new warnings
> * [ ]  I have added tests that prove my fix is effective or that my feature works
> * [x]  New and existing unit tests pass locally with my changes
> * [ ]  Any dependent changes have been merged and published in downstream modules


> # Description
> I designed the password generator ui and it works fine without bug but when i updated my folked repository and pull it into my mechine i get this error aside that my implementation is working great Screenshot from 2022-05-30 03-06-39
> 
> Fixes # (issue)
> 
> ## Type of change
> * [x]  New feature (non-breaking change which adds functionality)
> 
> # How Has This Been Tested?
> Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
> 
> # Checklist:
> * [x]  I have performed a self-review of my own code
> * [x]  I have commented my code, particularly in hard-to-understand areas
> * [ ]  I have made corresponding changes to the documentation
> * [ ]  My changes generate no new warnings
> * [ ]  I have added tests that prove my fix is effective or that my feature works
> * [x]  New and existing unit tests pass locally with my changes
> * [ ]  Any dependent changes have been merged and published in downstream modules

@vasantisuthar could you please help?

**Comment by atapas** [comment]
@Asterisk-z 

You need to do `npm install` or `yarn` again as the incoming changes may introduce the latest dependencies.

So please stop the local server, install/update and rerun it. That should solve it.

**Comment by Asterisk-z** [comment]
> @Asterisk-z
> 
> You need to do `npm install` or `yarn` again as the incoming changes may introduce the latest dependencies.
> 
> So please stop the local server, install/update and rerun it. That should solve it.

Okay i will try that

**Comment by Asterisk-z** [comment]
@atapas It works fine now

**Comment by atapas** [comment]
Hello @Asterisk-z 

We will be moving the react-play project from the `atapas` account to an organization to help manage the project in a better way. The new organization URL is: https://github.com/reactplay.

> **The movement will take place between 12:30 - 14:00 hrs IST(7:00 am - 8:30 am GMT), 30th May 2022. Please refrain from pushing any changes between this time.**

We will inform you when the movement is complete. You need to make a few minimal configuration changes after that. We will let you know.

**Comment by atapas** [comment]
> Hello @Asterisk-z
> 
> We will be moving the react-play project from the `atapas` account to an organization to help manage the project in a better way. The new organization URL is: https://github.com/reactplay.
> 
> > **The movement will take place between 12:30 - 14:00 hrs IST(7:00 am - 8:30 am GMT), 30th May 2022. Please refrain from pushing any changes between this time.**
> 
> We will inform you when the movement is complete. You need to make a few minimal configuration changes after that. We will let you know.

The migration is complete. You can find the project here: https://github.com/reactplay

You no need to reclone the project. You just need to set the correct upstream. Here is the link: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-for-a-fork

**Comment by atapas** [comment]
@Asterisk-z is it ready for review?

File: src/plays/password-generator/PasswordGenerator.jsx [hunk 1]
```diff
@@ -24,22 +24,36 @@ function PasswordGenerator(props) {
...
+  const specialCharCheckBox = useRef(null)
+  const passwordField = useRef(null)
+
 
   // generating lowercase characters
   data.lowercase = data?.uppercase?.map((i) => i.toLowerCase());
 
   // generate a random number within limit which is provided
   const randomNumberGenerator = (limit) => {
+
```
Comment (atapas) [inline src/plays/password-generator/PasswordGenerator.jsx:39]: We can avoid unnecessary line breaks

File: src/plays/password-generator/PasswordGenerator.jsx [hunk 2]
```diff
@@ -58,89 +72,147 @@ function PasswordGenerator(props) {
...
-    setError(false);
-    const concated = arrangeData();
-    let finalPassword = "";
-    for (let i = 0; i < passwordConfig.length; i++) {
-      finalPassword += concated[randomNumberGenerator(concated.length)];
-    }
-    return finalPassword;
+
+      setError(false);
+
```
Comment (atapas) [inline src/plays/password-generator/PasswordGenerator.jsx:77]: There are lots of line breaks, lets avoid that,

**Review by atapas — CHANGES_REQUESTED** [review]
@Asterisk-z Could you please fix the responsiveness as well. The screen doesn't fit well on smaller screen devices.

**Comment by atapas** [comment]
@Angryman18

Could you please review this PR as well as the code was originated by you, it will be helpful. Thanks!

File: src/plays/password-generator/PasswordGenerator.jsx (hunk 2 above)
Comment (Asterisk-z) [reply src/plays/password-generator/PasswordGenerator.jsx:77]: line breaks and responsiveness 
Let me clean it up now and send another PR

**Comment by Angryman18** [comment]
> @Angryman18
> 
> Could you please review this PR as well as the code was originated by you, it will be helpful. Thanks!

yes. sure.

File: src/plays/password-generator/PasswordGenerator.jsx (hunk 2 above)
Comment (Asterisk-z) [reply src/plays/password-generator/PasswordGenerator.jsx]: I have cleaned the code and set up responsiveness

File: src/plays/password-generator/PasswordGenerator.jsx [hunk 3]
```diff
@@ -24,6 +24,11 @@ function PasswordGenerator(props) {
   const [password, setPassword] = useState({ status: false, password: "" });
   const [passwordConfig, setPasswordConfig] = useState({ ...config });
   const [error, setError] = useState(false);
+  const lowercaseCheckBox = useRef(null)
+  const uppercaseCheckBox = useRef(null)
+  const numberCheckBox = useRef(null)
+  const specialCharCheckBox = useRef(null)
+  const passwordField = useRef(null)
```
Comment (Angryman18) [inline src/plays/password-generator/PasswordGenerator.jsx:31]: why do we need refs here? we have config state which can handle the changes.

File: src/plays/password-generator/PasswordGenerator.jsx [hunk 4]
```diff
@@ -58,89 +61,107 @@ function PasswordGenerator(props) {
...
+        finalPassword += concated[randomNumberGenerator(concated.length)];
+      }
+      return finalPassword;
   };
 
   // generate password button click handler
   const generateHander = () => {
     const finalPassword = generatePassword();
     setPassword({ status: false, password: finalPassword });
+    passwordField.current.value = finalPassword;
```
Comment (Angryman18) [inline src/plays/password-generator/PasswordGenerator.jsx:77]: we already have the password in the password state so why we need to manage ref?

File: src/plays/password-generator/PasswordGenerator.jsx [hunk 5]
```diff
@@ -58,89 +61,107 @@ function PasswordGenerator(props) {
...
 
   useEffect(() => {
     generateHander();
   }, []);
 
-  const ErrorBox = () => {
-    return <p className='error'>You cannot Uncheck All At Once.</p>;
+  const setLength = (value) => {
+    setPasswordConfig({ ...passwordConfig, ['length']: value });
+    setError(false);
```
Comment (Angryman18) [inline src/plays/password-generator/PasswordGenerator.jsx:111]: the value is a string here. although the for loop works but this needs to be a number.

File: src/plays/password-generator/PasswordGenerator.jsx [hunk 6]
```diff
@@ -58,89 +61,107 @@ function PasswordGenerator(props) {
...
-                id='numbers'
-              />
-              <div className='checkbox-comp '>
-                <label>Length</label>
-                <select
+        <div className="main">
+          <h1 className="title">Password Generator</h1>
+          {error && <ErrorBox />}
+            <div className="inputfield">
+                <input type="text" name="" className="text" id="" disabled  ref={passwordField}/>
```
Comment (Angryman18) [inline src/plays/password-generator/PasswordGenerator.jsx:122]: we can remove the the element which doest contain any value like "id" and "name"

**Comment by atapas** [comment]
> @atapas @Asterisk-z its good now. i see an issue is not resolved yet. anyway we can omit it since the code works fine.

@Asterisk-z are you resolving it?

**Comment by atapas** [comment]
@Angryman18 Please approve the PR if you are good. That helps me to merge it.

**Review by atapas — APPROVED** [review]

**Review by koustov — APPROVED** [review]
lgtm

**Review by Angryman18 — APPROVED** [review]

**Comment by atapas** [comment]
@all-contributors please add @Asterisk-z  for Code

**Pull request merged**